### PR TITLE
decompile egg hatch

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

File: sym_common.txt [hunk 1]
```diff
@@ -110,7 +110,7 @@ gUnknown_0300434C: @ 300434C
 @ egg_hatch
 
 	.align 2
-gUnknown_0300481C: @ 300481C
+sEggHatchData: @ 300481C
```
Comment (camthesaxman) [inline sym_common.txt]: Common syms are not static, so it would be better to name this `gEggHatchData`

File: src/egg_hatch.c [hunk 2]
```diff
@@ -50,3 +268,596 @@ void CreatedHatchedMon(struct Pokemon *egg, struct Pokemon *temp) {
...
+            sprite->pos2.x = Sin(sprite->data1, 2);
+            if (sprite->data0 == 15)
+            {
+                PlaySE(23);
+                StartSpriteAnim(sprite, 2);
+                CreateRandomEggShardSprite();
+                CreateRandomEggShardSprite();
+            }
+            if (sprite->data0 == 30)
+                PlaySE(23);
```
Comment (camthesaxman) [inline src/egg_hatch.c]: `SE_BOWA`

Comment (DizzyEggg) [inline src/egg_hatch.c]: changed

File: sym_common.txt (hunk 1 above)
Comment (DizzyEggg) [inline sym_common.txt]: Renamed and created another file in the common syms folder

File: src/egg_hatch.c [hunk 3]
```diff
@@ -50,3 +269,596 @@ void CreatedHatchedMon(struct Pokemon *egg, struct Pokemon *temp) {
...
+        {
+            MenuDrawTextWindow(22, 8, 27, 13);
+            InitYesNoMenu(22, 8, 4);
+            gEggHatchData->CB2_state++;
+        }
+        break;
+    case 9:
+        {
+            s8 menuInput;
+            if ((menuInput = ProcessMenuInputNoWrap_()) != -2)
```
Comment (camthesaxman) [inline src/egg_hatch.c]: Is there any more readable way to write this? Maybe a switch?

Comment (DizzyEggg) [inline src/egg_hatch.c]: I tried a switch but it didn't work either.

Comment (camthesaxman) [inline src/egg_hatch.c]: This matches, if you think it's more readable.
`            if ((menuInput = ProcessMenuInputNoWrap_()) != -2)
            {
                if (menuInput == -1 || menuInput == 1)
                {
                    gEggHatchData->CB2_state++;
                }
                else
                {
                    u16 species;
                    u8 gender;
                    u32 personality;

                    GetMonNick(&gPlayerParty[gEggHatchData->eggPartyID], gStringVar3);
                    species = GetMonData(&gPlayerParty[gEggHatchData->eggPartyID], MON_DATA_SPECIES);
                    gender = GetMonGender(&gPlayerParty[gEggHatchData->eggPartyID]);
                    personality = GetMonData(&gPlayerParty[gEggHatchData->eggPartyID], MON_DATA_PERSONALITY, 0);
                    DoNamingScreen(3, gStringVar3, species, gender, personality, EggHatchSetMonNickname);
                }
            }`

**Pull request merged**